### ISSUE TITLE
How to use training and testing methods in original paper to reproduce the accuracy in original paper？

### ISSUE THREAD

**Issue by Tracisran** [CoinCheung/BiSeNet#220]
Hi,

I‘m facing some problems in reproducing the acc declared in the origin paper of bisenetv2.
In the original paper, for cityscapes val and test set, the input size is resized into half (512*1024) for inference then the output size is resized to the original image for calculate the mIoU. Besides, in data augment resize range is (0.25 , 1) and no color jitter is used. I used these hyperparameters to train the model in your framework, but I got only 71.1 mIou for valset and 69.0 for test set. It's too far from the results declared by the author (72.6 for test set). Then i added the colorjitter in it, the results got better but it was still toooo low. The same phenomenon occurred  in framework Paddleseg. 

So, I am confused how can I reproduce the results of the paper.

In addition, another question is bothering me that why for cityscapes, test mIoU is much lower than val mIoU, the IoU for bus and truck in test is almost lower than val by 0.2. This is a common phenomenon for many light models that I reproduced. But in their papers, almost no difference between val and test.

It's so kind of you to handle my problems！

**Comment by CoinCheung** [comment]
Hi,

Two reasons I can think of now why you cannot reproduce the result in paper:
1. the essential reason is that the author did not describe the exact details of the model in paper. I implement this based on my understanding.
2. The models on cityscapes have large variance. The result varies within a big interval. So it is better to training for 5 times, and compute the mean/var of the mious. 

As for ` why for cityscapes, test mIoU is much lower than val mIoU, the IoU for bus and truck in test is almost lower than val by 0.2. This is a common phenomenon for many light models that I reproduced`, My understanding is this: Because the light-weighted models ususally use longer training scheduler(bisenetv2 uses 150k iterations) and cityscapes has too few pictures,  thus the model tends to overfit on this dataset, which hinders the generalization of the model from val set to test set.

**Comment by Tracisran** [comment]
Thanks for your response。
I used multi-step schedule to train the model for times, this schedule could get results which are more stable. After Bisenetv2, i trained DSTL which got much higher results in paper using codes authors provided. i did not use pretrained weights (at least i think in the paper the author doesn't say that pretrained weights are used ), the results are worse than bisenetv2. I cannot understand.

The second question, I have observed that the distributions of test and val sets are very different on the bus and truck classes, so I just can't understand why comparable results can be achieved in many papers.
